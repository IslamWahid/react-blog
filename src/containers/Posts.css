.Posts {
  padding-bottom: 15px;
}
