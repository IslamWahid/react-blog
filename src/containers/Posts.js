import React, { Component } from 'react';
import { Panel } from 'react-bootstrap';
import axios from 'axios';
import './Posts.css';

export default class Posts extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: null,
      isDeleting: null,
      post: null,
      title: '',
      body: ''
    };
  }

  async componentDidMount() {
    try {
      const { data: post } = await axios.get(
        `/posts/${this.props.match.params.id}`
      );
      const { title, body } = post;

      this.setState({
        post,
        title,
        body
      });
    } catch (e) {
      alert(e);
    }
  }

  validateForm() {
    return this.state.title.length > 0 && this.state.body.length > 0;
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();

    this.setState({ isLoading: true });
  };

  handleDelete = async event => {
    event.preventDefault();

    const confirmed = window.confirm(
      'Are you sure you want to delete this post?'
    );

    if (!confirmed) {
      return;
    }

    this.setState({ isDeleting: true });
  };

  render() {
    return (
      <div className="Posts">
        {this.state.post && (
          <Panel>
            <Panel.Heading>
              <Panel.Title componentClass="h3">{this.state.title}</Panel.Title>
            </Panel.Heading>
            <Panel.Body>{this.state.body}</Panel.Body>
          </Panel>
        )}
      </div>
    );
  }
}
