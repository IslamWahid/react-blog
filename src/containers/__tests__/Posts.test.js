import React from 'react';
import Posts from '../Posts';
import { render } from 'enzyme';

describe('Posts Component', () => {
  let posts;

  beforeEach(() => {
    posts = render(<Posts path="/posts/:id" exact />);
  });
  it('should match snapshot', () => {
    expect(posts).toMatchSnapshot();
  });
});
