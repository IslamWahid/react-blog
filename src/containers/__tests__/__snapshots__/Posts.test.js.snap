// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Component should match snapshot 1`] = `
initialize {
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
